tools/src/bin/rust-indexer.rs

```diff
@@ -32,12 +32,42 @@ pub struct Defs {
     map: HashMap<DefId, data::Def>,
 }
 
+/// Local filesystem path mappings and metadata which exist for the following
+/// purposes:
+/// 1. Know where to output the analysis files.
+///   - There is only ever one analysis output directory.
+/// 2. Know how to locate rust source files in order to hackily extract strings
+///    that should have been in the save-analysis files.
+///    - There are potentially 4 source directories: the real source directory
+///      (cross-platform), the per-platform objdir where generated source files
+///      live in practice, the synthetic unified objdir that the
+///      merge-analyses steps creates out of the per-platform objdirs, and the
+///      rust stdlib directory.
+///    - We ignore the concept of the unified objdir and hope that all files
+///      be identical at a source level so they merge cleanly.  If they don't,
+///      it's assumed merge-analyses will perform any necessary fix-ups.
 #[derive(Debug)]
 struct TreeInfo<'a> {
-    src_dir: &'a Path,
-    output_dir: &'a Path,
-    objdir: &'a Path,
+    /// Local filesystem path root for the analysis dir where rust-indexer.rs
+    /// should write its output.
+    out_analysis_dir: &'a Path,
+    /// Local filesystem path root for the source tree.  In the searchfox path
+    /// space presented to users, this means all paths not prefixed with
+    /// `__GENERATED__`.
+    srcdir: &'a Path,
+    /// Local filesystem path root for the per-platform generated source tree.
+    /// In the searchfox path space presented to users, this means paths
+    /// prefixed with `__GENERATED__`.
+    generated: &'a Path,
+    /// The searchfox path space prefix for generated.
+    generated_friendly: &'a Path,
+    /// Local filesystem path root for the rust stdlib source.  In the searchfox
+    /// path space presented to users, this means paths prefixed with
+    /// `__GENERATED__/__RUST__`.
     libstd: &'a Path,
+    /// The searchfox path space prefix for the rust stdlib, assumed to be a
+    /// suffix of objdir_friendly.
```

Comment:
Assuming this comment should refer to `generated_friendly` instead of `objdir_friendly`. Also, not sure if suffix is the right word here. Extension, maybe? Or maybe reword it to say `generated_friendly` is a prefix of this?

Reply:
Yes, your generated_friendly phrasing is much more clear, thank you.  (And indeed there was a late rename from objdir_friendly to generated_friendly that missed this.)